packages/widget-v2/src/hooks/useCreateEvmWallets.tsx

```diff
@@ -0,0 +1,94 @@
+import { seiPrecompileAddrABI } from "@/constants/abis";
+import { evmWaletAtom, MinimalWallet } from "@/state/wallets";
+import { useSetAtom } from "jotai";
+import { createPublicClient, http } from "viem";
+import { sei } from "viem/chains";
+import { useAccount as useEvmAccount, useDisconnect as useEvmDisconnect, useConnect as useEvmConnect } from "wagmi";
+
+
+export const useCreateEvmWallets = () => {
+  const setEvmWallet = useSetAtom(evmWaletAtom);
+  const {
+    connector: currentEvmConnector, address: evmAddress, isConnected: isEvmConnected,
+  } = useEvmAccount();
+  const { disconnectAsync } = useEvmDisconnect();
+  const { connectors, connectAsync } = useEvmConnect();
+
+  const createEvmWallets = (chainID: string) => {
+    const isSei = chainID === "pacific-1";
+
+    const wallets: MinimalWallet[] = [];
+    for (const connector of connectors) {
+      const evmGetAddress: MinimalWallet["getAddress"] = async ({
+        signRequired,
+      }) => {
+        if (connector.id !== currentEvmConnector?.id) {
+          await connectAsync({ connector, chainId: Number(chainID) });
+          setEvmWallet({ walletName: connector.id, chainType: "evm" });
+        } else if (evmAddress && isEvmConnected && signRequired) {
+          setEvmWallet({ walletName: connector.id, chainType: "evm" });
+        }
+        return evmAddress;
+      };
+
+      const minimalWallet: MinimalWallet = {
+        walletName: connector.id,
+        walletPrettyName: connector.name,
+        walletChainType: "evm",
+        walletInfo: {
+          logo: connector.icon,
+        },
+        connect: async () => {
+          if (connector.id === currentEvmConnector?.id) return;
+          try {
+            await connectAsync({ connector, chainId: Number(chainID) });
+            setEvmWallet({ walletName: connector.id, chainType: "evm" });
+            // TODO: onWalletConnected
+          } catch (error) {
+            console.error(error);
+            throw error;
```

Comment:
same with this one, we should test what happens if this error is thrown

Reply:
yeah I'm still figuring out how we going to display these errors without toast